### PR TITLE
Upgrading to use AWS SDK v2

### PR DESCRIPTION
**Pull request by Tang8330**
As per title. Let's get this merged in after our existing work on DynamoDB and MongoDB streaming.

Things to test manually:

- [x] DynamoDB Streaming
- [x] DynamoDB Snapshot

### CODE REVIEW

File: lib/dynamo/message.go [hunk 1]
```diff
@@ -26,77 +26,64 @@ func stringToFloat64(s string) (float64, error) {
...
-		// Convert the string set to a slice of strings.
-		strSet := make([]string, len(attr.SS))
-		for i, s := range attr.SS {
-			strSet[i] = *s
-		}
-
+	case *types.AttributeValueMemberSS:
+		strSet := make([]string, len(v.Value))
+		copy(strSet, v.Value)
 		return strSet, nil
```
Comment (nathan-artie) [inline lib/dynamo/message.go:64]: Can possibly do `return slices.Clone(v.Value), nil`

File: lib/dynamo/parse_message_test.go [hunk 2]
```diff
@@ -1,103 +1,98 @@
...
 				ApproximateCreationDateTime: aws.Time(time.Date(2023, 8, 28, 0, 0, 0, 0, time.UTC)),
 			},
-			EventName: ptr.ToString("INSERT"),
+			EventName: types.OperationTypeInsert,
 		}, "testTable")
 
 		assert.NoError(t, err)
-		assert.Equal(t, map[string]any{"foo": "bar"}, msg.afterRowData)
-		assert.Equal(t, map[string]any{"user_id": "123"}, msg.primaryKey)
+		assert.Equal(t, map[string]interface{}{"foo": "bar"}, msg.afterRowData)
```
Comment (nathan-artie) [inline lib/dynamo/parse_message_test.go:42]: Did these need to change from `interface{}`?

File: scripts/dynamo/load.go [hunk 3]
```diff
@@ -30,81 +31,73 @@ func main() {
 		logger.Fatal("Please provide a valid number for rows")
 	}
 
-	sess, err := session.NewSession(&aws.Config{
-		Region: aws.String(region),
-	})
+	awsCfg, err := config.LoadDefaultConfig(context.TODO(),
```
Comment (nathan-artie) [inline scripts/dynamo/load.go:34]: `context.Background()`?

**Review by nathan-artie — APPROVED** [review]

File: lib/dynamo/parse_message.go [hunk 4]
```diff
@@ -2,22 +2,20 @@ package dynamo
 
 import (
 	"fmt"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodbstreams/types"
```
Comment (nathan-artie) [inline lib/dynamo/parse_message.go:5]: Import order.

File: lib/s3lib/dynamodb_export.go [hunk 5]
```diff
@@ -0,0 +1,192 @@
+package s3lib
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodbstreams/types"
+)
+
+func assertType[T any](val interface{}) (T, error) {
```
Comment (nathan-artie) [inline lib/s3lib/dynamodb_export.go:10]: `any`.

File: lib/s3lib/dynamodb_export.go [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+	castedVal, isOk := val.(T)
+	if !isOk {
+		var zero T
+		return zero, fmt.Errorf("expected type %T, got %T", zero, val)
+	}
+	return castedVal, nil
+}
+
+type exportedPayload struct {
+	Item map[string]map[string]interface{} `json:"Item"`
```
Comment (nathan-artie) [inline lib/s3lib/dynamodb_export.go:20]: `any`.

File: lib/s3lib/dynamodb_export.go [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+			}
+			return &types.AttributeValueMemberM{Value: m}, nil
+		case "L":
+			castedList, isOk := val.([]any)
+			if !isOk {
+				return nil, fmt.Errorf("expected []any{}, got: %T", val)
+			}
+
+			list := make([]types.AttributeValue, len(castedList))
+			for i, lv := range val.([]interface{}) {
```
Comment (nathan-artie) [inline lib/s3lib/dynamodb_export.go:170]: `any`.

Comment (nathan-artie) [reply lib/s3lib/dynamodb_export.go:170]: Range over `castedList` instead of `val.([]interface{})`?

Comment (Tang8330) [reply lib/s3lib/dynamodb_export.go:170]: yup, good flag

**Review by nathan-artie — APPROVED** [review]

**Pull request merged**